### PR TITLE
Fix OptionsValidator source-gen to skip static and const members

### LINKED ISSUE #88150: OptionsValidator source-gen shouldn't cover static and const members
### Description

Recently added OptionsValidator (issue #85475 and PR #87587) considers all members of a `TOption` for validation, however we shouldn't validate `static` ones. Additionally, `const` fields should be ignored or a warning should be emitted if such members are annotated with data validation attributes (inherited from `ValidationAttribute`).

### Reproduction Steps

```csharp
[OptionsValidator]
public partial class MyValidator : IValidateOptions<MyOptions>
{
}

public class MyOptions
{
    [Required]
    public static string? StaticStringField;

    [Required]
    public static string? StaticStringProperty => "static";

    [Required]
    public const string? ConstString = null;
}
```

### Expected behavior

A warning is emitted for all three members, no `static` or `const` members are validated in the generated code.

### Actual behavior

The generator produces the next code which won't compile:
image

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

.NET SDK 8.0.100-preview.7.23327.3

### Other information

_No response_

### PR DESCRIPTION
**Pull request by xakep139**
Fix #88150.
Also, performance of the generator was improved - now it doesn't check all members of `string` type.

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Options/Microsoft.Extensions.Options.sln
Comment (xakep139) [inline src/libraries/Microsoft.Extensions.Options/Microsoft.Extensions.Options.sln:1]: I wasn't sure whether I can edit solution file or not

File: src/libraries/Microsoft.Extensions.Options/gen/Parser.cs [hunk 1]
```diff
@@ -462,7 +476,7 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
             }
 
             // generate a warning if the field/property seems like it should be enumerated
-            if (enumerationValidatorTypeName == null && speculate)
+            if (enumerationValidatorTypeName == null && speculate && memberType.SpecialType != SpecialType.System_String)
```
Comment (xakep139) [inline src/libraries/Microsoft.Extensions.Options/gen/Parser.cs:479]: We don't check `string` internals anymore

File: src/libraries/Microsoft.Extensions.Options/gen/Resources/xlf/Strings.ru.xlf [hunk 2]
```diff
@@ -59,7 +69,7 @@
       </trans-unit>
       <trans-unit id="MemberIsInaccessibleTitle">
         <source>Can't validate private fields or properties.</source>
-        <target state="translated">Не удается проверить частные поля или свойства.</target>
+        <target state="translated">Не удается проверить приватные поля или свойства.</target>
```
Comment (xakep139) [inline src/libraries/Microsoft.Extensions.Options/gen/Resources/xlf/Strings.ru.xlf:72]: This fixes an incorrect translation.

File: src/libraries/Microsoft.Extensions.Options/gen/Parser.cs [hunk 3]
```diff
@@ -448,6 +448,20 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
                 }
             }
 
+            bool validationAttributeIsApplied = validationAttrs.Count > 0 || transValidatorTypeName is not null || enumerationValidatorTypeName is not null;
+
+            if (member.IsStatic)
+            {
+                // generate an error if the member is static and has a validation attribute applied
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Options/gen/Parser.cs:455]: > // generate an error if the member is static and has a validation attribute applied [](http://example.com/codeflow?start=15&length=84)

Is it the correct thing to generate error and not just ignoring the statics?

File: src/libraries/Microsoft.Extensions.Options/Microsoft.Extensions.Options.sln
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/Microsoft.Extensions.Options.sln:1]: As long as the solution loads and works, that is ok.

**Comment by tarekgh** [comment]
Looks you missed adding one resources 

```
src/libraries/Microsoft.Extensions.Options/gen/DiagDescriptors.cs(97,23): error CS0117: (NETCORE_ENGINEERING_TELEMETRY=Build) 'SR' does not contain a definition for 'CantValidateStaticOrConstMemberTitle'
```

File: src/libraries/Microsoft.Extensions.Options/gen/Parser.cs (hunk 3 above)
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/gen/Parser.cs:455]: Would be good if we check what is the runtime behavior when having the validation attribute on the statics.

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/gen/Parser.cs:455]:
```
using a validation framework like ASP.NET MVC or ASP.NET Core, the RequiredAttribute alone would not automatically trigger validation for static properties. These frameworks primarily focus on validating instance properties of classes that are involved in HTTP request processing.
```

Considering that, I would say making it as `warning` will make sense. 

@jeffhandley @geeknoid what do you think about that?

**Comment by xakep139** [comment]
> Looks you missed adding one resources
> 
> ```
> src/libraries/Microsoft.Extensions.Options/gen/DiagDescriptors.cs(97,23): error CS0117: (NETCORE_ENGINEERING_TELEMETRY=Build) 'SR' does not contain a definition for 'CantValidateStaticOrConstMemberTitle'
> ```

And that's strange, because you can see I added it into `src/libraries/Microsoft.Extensions.Options/gen/Resources/Strings.resx`

**Comment by tarekgh** [comment]
The error is coming from the project `Microsoft.Extensions.Options.SourceGeneration.Tests.csproj`. You need to add it to https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Options/tests/SourceGenerationTests/Resources/Strings.resx too. It is in my to do list to consolidate the resource files into global one.

File: src/libraries/Microsoft.Extensions.Options/gen/Parser.cs (hunk 3 above)
Comment (xakep139) [reply src/libraries/Microsoft.Extensions.Options/gen/Parser.cs]: Turns out it was a warning already (bad copy-paste). Added an additional assertion in tests to ensure that

**Review by tarekgh — APPROVED** [review]
Thanks @xakep139 providing the changes. LGTM

**Pull request merged**